sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeLocalShuffleReader.scala

```diff
@@ -72,10 +72,24 @@ object OptimizeLocalShuffleReader extends CustomShuffleReaderRule {
     assert(numMappers > 0)
     val numReducers = shuffleStage.shuffle.numPartitions
     val expectedParallelism = advisoryParallelism.getOrElse(numReducers)
-    val splitPoints = equallyDivide(numReducers, math.max(1, expectedParallelism / numMappers))
-    (0 until numMappers).flatMap { mapIndex =>
-      (splitPoints :+ numReducers).sliding(2).map {
-        case Seq(start, end) => PartialMapperPartitionSpec(mapIndex, start, end)
+    val splitPoints = if (numMappers == 0) {
```

Comment:
we don't need this check anymore, see line 72.

Reply:
good point!